### PR TITLE
Add EventGrid MQTT, App Config events

### PR DESCRIPTION
**Pull request by billwert**
Also includes a rewrite of some of my autorest customization that was prone to races.

### CODE REVIEW

File: sdk/eventgrid/azure-messaging-eventgrid/swagger/src/main/java/EventGridCustomization.java [hunk 1]
```diff
@@ -229,63 +236,86 @@ public void customizeResourceEvents(LibraryCustomization customization, Logger l
...
+                    compilationUnit.addImport("com.azure.core.util.serializer.SerializerEncoding");
+                    compilationUnit.addImport("java.io.IOException");
+                    compilationUnit.addImport("java.io.UncheckedIOException");
+                    classDec.addFieldWithInitializer("ClientLogger", "LOGGER", parseExpression("new ClientLogger(" + className + ".class)"), Keyword.STATIC, Keyword.FINAL, Keyword.PRIVATE);
+                    classDec.addFieldWithInitializer("SerializerAdapter", "DEFAULT_SERIALIZER_ADAPTER", parseExpression("JacksonAdapter.createDefaultSerializerAdapter()"), Keyword.STATIC, Keyword.FINAL, Keyword.PRIVATE);
+
+
+                    Arrays.asList("Authorization", "Claims", "HttpRequest").forEach(method -> {
+                        classDec.getMethodsByName("get" + method).get(0).setName("getResource" + method);
+                        classDec.getMethodsByName("set" + method).get(0).setName("setResource" + method);
```
Comment (alzimmermsft) [inline sdk/eventgrid/azure-messaging-eventgrid/swagger/src/main/java/EventGridCustomization.java:256]: May want to use a slightly safer design like

```java
classDec.getMethodByName("get" + method).ifPresent(getter -> getter.setName("getResource" + method));
classDec.getMethodByName("set" + method).ifPresent(setter -> setter.setName("setResource" + method));
```

**Review by alzimmermsft — APPROVED** [review]

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+     * The sync token representing the server state after the event.
+     */
+    @JsonProperty(value = "syncToken")
+    private String syncToken;
+
+    /** Creates an instance of AppConfigurationSnapshotCreatedEventData class. */
+    public AppConfigurationSnapshotCreatedEventData() {}
+
+    /**
+     * Get the name property: The name of snapshot that was created.
```
Comment (joshfree) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java:35]: nit: should be `The name of the snapshot that was created.`

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+     *
+     * @return the name value.
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * Set the name property: The name of snapshot that was created.
```
Comment (joshfree) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java:44]: ``The name of the snapshot that was created.`

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotModifiedEventData.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+package com.azure.messaging.eventgrid.systemevents;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** Schema of the Data property of an EventGridEvent for a Microsoft.AppConfiguration.SnapshotModified event. */
+@Fluent
+public final class AppConfigurationSnapshotModifiedEventData {
+    /*
+     * The name of snapshot that was modified.
```
Comment (joshfree) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotModifiedEventData.java:14]: missing `the`

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotModifiedEventData.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+     * The sync token representing the server state after the event.
+     */
+    @JsonProperty(value = "syncToken")
+    private String syncToken;
+
+    /** Creates an instance of AppConfigurationSnapshotModifiedEventData class. */
+    public AppConfigurationSnapshotModifiedEventData() {}
+
+    /**
+     * Get the name property: The name of snapshot that was modified.
```
Comment (joshfree) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotModifiedEventData.java:35]: `the`

**Review by joshfree — APPROVED** [review]

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/ResourceActionFailureEventData.java [hunk 6]
```diff
@@ -341,7 +341,7 @@ public String getClaims() {
      * @deprecated This method is no longer supported since v4.9.0.
      *     <p>Use {@link ResourceActionFailureEventData#setResourceClaims(Map)} instead.
      */
-    @Deprecated
+    @Deprecated()
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/ResourceActionFailureEventData.java:344]: Is there a reason for adding `()` to this annotation?

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java (hunk 2 above)
Comment (billwert) [reply sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java:35]: good catch!

File: sdk/eventgrid/azure-messaging-eventgrid/swagger/src/main/java/EventGridCustomization.java (hunk 1 above)
Comment (srnagar) [reply sdk/eventgrid/azure-messaging-eventgrid/swagger/src/main/java/EventGridCustomization.java:256]: There is no `getMethodByName` in `classDec`. This returns a list and we should ensure all methods contained in that list get renamed as we may add overloads. So, this should be more like:

`classDec.getMethodsByName("oldName").forEach(methodDeclaration -> methodDeclaration.setName("newName")`

**Review by srnagar — APPROVED** [review]

File: sdk/eventgrid/azure-messaging-eventgrid/swagger/src/main/java/EventGridCustomization.java (hunk 1 above)
Comment (billwert) [reply sdk/eventgrid/azure-messaging-eventgrid/swagger/src/main/java/EventGridCustomization.java]: done! good call.

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/ResourceActionFailureEventData.java (hunk 6 above)
Comment (billwert) [reply sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/ResourceActionFailureEventData.java]: It's gone now - I had fixed it but just not pushed the branch.

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java (hunk 2 above)
Comment (billwert) [reply sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AppConfigurationSnapshotCreatedEventData.java:35]: Getting a swagger update for this one.

**Comment by billwert** [comment]
@srnagar Just calling out the additional change for a new property and a string change since your approval.

**Review by srnagar — APPROVED** [review]
LGTM!

**Pull request merged**